python3.9 compatibility

**Describe the bug**
Arch Linux is currently rebuilding all python related things against python3.9.
The python-fastnumbers package has failing tests on python3.9 and looking at the build output I am not entirely sure they can just be ignored.

**Expected behavior**
python3.9 is supported

**Environment (please complete the following information):**
 - fastnumbers version: 3.0.0 
 - Python Version: 3.9.0
 - OS: Arch Linux
 - Compiler (if known): gcc 10.2.0

**To Reproduce**

```
python setup.py build
local _py3_ver=$(python --version | cut -d " " -f2)
export PYTHONPATH="build/lib.linux-$CARCH-${_py3_ver%"."*}:${PYTHONPATH}"
pytest -v
```

```
running build
running build_ext
building 'fastnumbers' extension
creating build
creating build/temp.linux-x86_64-3.9
creating build/temp.linux-x86_64-3.9/src
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -fPIC -I/build/python-fastnumbers/src/python-fastnumbers-3.0.0/include -I/usr/include/python3.9 -c src/fastnumbers.c -o build/temp.linux-x86_64-3.9/src/fastnumbers.o
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -fPIC -I/build/python-fastnumbers/src/python-fastnumbers-3.0.0/include -I/usr/include/python3.9 -c src/numbers.c -o build/temp.linux-x86_64-3.9/src/numbers.o
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -fPIC -I/build/python-fastnumbers/src/python-fastnumbers-3.0.0/include -I/usr/include/python3.9 -c src/objects.c -o build/temp.linux-x86_64-3.9/src/objects.o
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -fPIC -I/build/python-fastnumbers/src/python-fastnumbers-3.0.0/include -I/usr/include/python3.9 -c src/parsing.c -o build/temp.linux-x86_64-3.9/src/parsing.o
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -fPIC -I/build/python-fastnumbers/src/python-fastnumbers-3.0.0/include -I/usr/include/python3.9 -c src/strings.c -o build/temp.linux-x86_64-3.9/src/strings.o
In file included from src/strings.c:14:
src/strings.c: In function ‘str_to_PyFloat’:
/build/python-fastnumbers/src/python-fastnumbers-3.0.0/include/fastnumbers/numbers.h:24:55: warning: implicit declaration of function ‘_Py_dg_stdnan’ [-Wimplicit-function-declaration]
   24 | #define PyFloat_from_NaN(negative) PyFloat_FromDouble(_Py_dg_stdnan(negative));
      |                                                       ^~~~~~~~~~~~~
src/strings.c:115:20: note: in expansion of macro ‘PyFloat_from_NaN’
  115 |             return PyFloat_from_NaN(sign < 0);
      |                    ^~~~~~~~~~~~~~~~
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fno-semantic-interposition -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 -fPIC -I/build/python-fastnumbers/src/python-fastnumbers-3.0.0/include -I/usr/include/python3.9 -c src/unicode_character.c -o build/temp.linux-x86_64-3.9/src/unicode_character.o
creating build/lib.linux-x86_64-3.9
gcc -pthread -shared -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fno-semantic-interposition -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.9/src/fastnumbers.o build/temp.linux-x86_64-3.9/src/numbers.o build/temp.linux-x86_64-3.9/src/objects.o build/temp.linux-x86_64-3.9/src/parsing.o build/temp.linux-x86_64-3.9/src/strings.o build/temp.linux-x86_64-3.9/src/unicode_character.o -L/usr/lib -o build/lib.linux-x86_64-3.9/fastnumbers.cpython-39-x86_64-linux-gnu.so

============================= test session starts ==============================
platform linux -- Python 3.9.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1 -- /usr/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/build/python-fastnumbers/src/python-fastnumbers-3.0.0/.hypothesis/examples')
rootdir: /build/python-fastnumbers/src/python-fastnumbers-3.0.0
plugins: hypothesis-5.41.2
collecting ... collected 233 items

tests/test_builtins.py::GeneralFloatCases::test_error_message PASSED     [  0%]
tests/test_builtins.py::GeneralFloatCases::test_float PASSED             [  0%]
tests/test_builtins.py::GeneralFloatCases::test_float_memoryview PASSED  [  1%]
tests/test_builtins.py::GeneralFloatCases::test_float_with_comma SKIPPED [  1%]
tests/test_builtins.py::GeneralFloatCases::test_floatconversion PASSED   [  2%]
tests/test_builtins.py::GeneralFloatCases::test_keyword_args PASSED      [  2%]
tests/test_builtins.py::GeneralFloatCases::test_non_numeric_input_types PASSED [  3%]
tests/test_builtins.py::GeneralFloatCases::test_underscores PASSED       [  3%]
tests/test_builtins.py::InfNanTest::test_inf_as_str PASSED               [  3%]
tests/test_builtins.py::InfNanTest::test_inf_from_str PASSED             [  4%]
tests/test_builtins.py::InfNanTest::test_inf_signs PASSED                [  4%]
tests/test_builtins.py::InfNanTest::test_nan_as_str PASSED               [  5%]
tests/test_builtins.py::InfNanTest::test_nan_from_str FAILED             [  5%]
tests/test_builtins.py::InfNanTest::test_nan_signs FAILED                [  6%]
tests/test_builtins.py::IntTestCases::test_basic PASSED                  [  6%]
tests/test_builtins.py::IntTestCases::test_error_message PASSED          [  6%]
tests/test_builtins.py::IntTestCases::test_int_base_bad_types PASSED     [  7%]
tests/test_builtins.py::IntTestCases::test_int_base_indexable PASSED     [  7%]
tests/test_builtins.py::IntTestCases::test_int_base_limits PASSED        [  8%]
tests/test_builtins.py::IntTestCases::test_int_memoryview PASSED         [  8%]
tests/test_builtins.py::IntTestCases::test_int_returns_int_subclass PASSED [  9%]
tests/test_builtins.py::IntTestCases::test_int_subclass_with_index PASSED [  9%]
tests/test_builtins.py::IntTestCases::test_int_subclass_with_int PASSED  [  9%]
tests/test_builtins.py::IntTestCases::test_intconversion PASSED          [ 10%]
tests/test_builtins.py::IntTestCases::test_issue31619 PASSED             [ 10%]
tests/test_builtins.py::IntTestCases::test_keyword_args PASSED           [ 11%]
tests/test_builtins.py::IntTestCases::test_no_args PASSED                [ 11%]
tests/test_builtins.py::IntTestCases::test_non_numeric_input_types PASSED [ 12%]
tests/test_builtins.py::IntTestCases::test_small_ints PASSED             [ 12%]
tests/test_builtins.py::IntTestCases::test_string_float PASSED           [ 12%]
tests/test_builtins.py::IntTestCases::test_underscores PASSED            [ 13%]
tests/test_fastnumbers.py::test_version PASSED                           [ 13%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[fast_real] PASSED [ 14%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[fast_float] PASSED [ 14%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[fast_int] PASSED [ 15%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[fast_forceint] PASSED [ 15%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[isreal] PASSED [ 15%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[isfloat] PASSED [ 16%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[isint] PASSED [ 16%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[isintlike] PASSED [ 17%]
tests/test_fastnumbers.py::test_invalid_argument_raises_type_error[real] PASSED [ 17%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[fast_real] PASSED [ 18%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[fast_float] PASSED [ 18%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[fast_int] PASSED [ 18%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[fast_forceint] PASSED [ 19%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[isreal] PASSED [ 19%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[isfloat] PASSED [ 20%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[isint] PASSED [ 20%]
tests/test_fastnumbers.py::test_no_arguments_raises_type_error[isintlike] PASSED [ 21%]
tests/test_fastnumbers.py::test_key_backwards_compatibility[fast_real] PASSED [ 21%]
tests/test_fastnumbers.py::test_key_backwards_compatibility[fast_float] PASSED [ 21%]
tests/test_fastnumbers.py::test_key_backwards_compatibility[fast_int] PASSED [ 22%]
tests/test_fastnumbers.py::test_key_backwards_compatibility[fast_forceint] PASSED [ 22%]
tests/test_fastnumbers.py::test_real_no_arguments_returns_0 PASSED       [ 23%]
tests/test_fastnumbers.py::test_real_returns_same_as_fast_real PASSED    [ 23%]
tests/test_fastnumbers.py::test_numbers_with_underscores_converted_according_to_allow_underscores_option[fast_real] PASSED [ 24%]
tests/test_fastnumbers.py::test_numbers_with_underscores_converted_according_to_allow_underscores_option[fast_float] PASSED [ 24%]
tests/test_fastnumbers.py::test_numbers_with_underscores_converted_according_to_allow_underscores_option[fast_int] PASSED [ 24%]
tests/test_fastnumbers.py::test_numbers_with_underscores_converted_according_to_allow_underscores_option[fast_forceint] PASSED [ 25%]
tests/test_fastnumbers.py::test_numbers_with_underscores_identified_according_to_allow_underscores_option[isreal] PASSED [ 25%]
tests/test_fastnumbers.py::test_numbers_with_underscores_identified_according_to_allow_underscores_option[isfloat] PASSED [ 26%]
tests/test_fastnumbers.py::test_numbers_with_underscores_identified_according_to_allow_underscores_option[isint] PASSED [ 26%]
tests/test_fastnumbers.py::test_numbers_with_underscores_identified_according_to_allow_underscores_option[isintlike] PASSED [ 27%]
tests/test_fastnumbers.py::TestFastReal::test_with_coerce_given_dumb_class_responds_to_internal_valueerror PASSED [ 27%]
tests/test_fastnumbers.py::TestFastReal::test_given_float_returns_float PASSED [ 27%]
tests/test_fastnumbers.py::TestFastReal::test_given_float_returns_int_if_intlike_with_coerce PASSED [ 28%]
tests/test_fastnumbers.py::TestFastReal::test_given_float_returns_float_or_int_with_coerce PASSED [ 28%]
tests/test_fastnumbers.py::TestFastReal::test_given_nan_returns_nan PASSED [ 29%]
tests/test_fastnumbers.py::TestFastReal::test_given_nan_returns_sub_value PASSED [ 29%]
tests/test_fastnumbers.py::TestFastReal::test_given_inf_returns_inf PASSED [ 30%]
tests/test_fastnumbers.py::TestFastReal::test_given_inf_returns_sub_value PASSED [ 30%]
tests/test_fastnumbers.py::TestFastReal::test_given_float_string_returns_float PASSED [ 30%]
tests/test_fastnumbers.py::TestFastReal::test_given_float_string_returns_int_with_coerce_with_intlike PASSED [ 31%]
tests/test_fastnumbers.py::TestFastReal::test_given_nan_string_returns_nan FAILED [ 31%]
tests/test_fastnumbers.py::TestFastReal::test_with_nan_given_nan_string_returns_sub_value PASSED [ 32%]
tests/test_fastnumbers.py::TestFastReal::test_given_inf_string_returns_inf PASSED [ 32%]
tests/test_fastnumbers.py::TestFastReal::test_given_very_large_float_returns_inf_with_coerce_on_or_off PASSED [ 33%]
tests/test_fastnumbers.py::TestFastReal::test_with_inf_given_inf_string_returns_sub_value PASSED [ 33%]
tests/test_fastnumbers.py::TestFastReal::test_given_padded_float_strings_returns_float PASSED [ 33%]
tests/test_fastnumbers.py::TestFastReal::test_given_padded_nan_returns_nan FAILED [ 34%]
tests/test_fastnumbers.py::TestFastReal::test_given_padded_inf_string_returns_inf PASSED [ 34%]
tests/test_fastnumbers.py::TestFastReal::test_given_int_returns_int PASSED [ 35%]
tests/test_fastnumbers.py::TestFastReal::test_given_int_returns_int_with_coerce PASSED [ 35%]
tests/test_fastnumbers.py::TestFastReal::test_given_int_string_returns_int PASSED [ 36%]
tests/test_fastnumbers.py::TestFastReal::test_given_padded_int_string_returns_int PASSED [ 36%]
tests/test_fastnumbers.py::TestFastReal::test_given_unicode_digit_returns_int PASSED [ 36%]
tests/test_fastnumbers.py::TestFastReal::test_given_unicode_numeral_returns_float PASSED [ 37%]
tests/test_fastnumbers.py::TestFastReal::test_given_unicode_non_numeral_returns_as_is PASSED [ 37%]
tests/test_fastnumbers.py::TestFastReal::test_given_unicode_of_more_than_one_char_returns_as_is PASSED [ 38%]
tests/test_fastnumbers.py::TestFastReal::test_given_invalid_string_returns_string_as_is PASSED [ 38%]
tests/test_fastnumbers.py::TestFastReal::test_given_invalid_string_raises_valueerror_if_raise_on_invalid_is_true PASSED [ 39%]
tests/test_fastnumbers.py::TestFastReal::test_given_invalid_type_raises_typeerror PASSED [ 39%]
tests/test_fastnumbers.py::TestFastReal::test_returns_default_value_if_given_invalid_string PASSED [ 39%]
tests/test_fastnumbers.py::TestFastReal::test_raises_valueerror_if_raise_on_invalid_is_true_with_default PASSED [ 40%]
tests/test_fastnumbers.py::TestFastReal::test_returns_input_as_is_if_valid_and_on_fail_is_given PASSED [ 40%]
tests/test_fastnumbers.py::TestFastReal::test_returns_transformed_input_if_invalid_and_on_fail_is_given PASSED [ 41%]
tests/test_fastnumbers.py::TestFastFloat::test_given_dumb_class_responds_to_internal_valueerror PASSED [ 41%]
tests/test_fastnumbers.py::TestFastFloat::test_given_float_returns_float PASSED [ 42%]
tests/test_fastnumbers.py::TestFastFloat::test_given_nan_returns_nan PASSED [ 42%]
tests/test_fastnumbers.py::TestFastFloat::test_given_nan_returns_sub_value PASSED [ 42%]
tests/test_fastnumbers.py::TestFastFloat::test_given_inf_returns_inf PASSED [ 43%]
tests/test_fastnumbers.py::TestFastFloat::test_given_inf_returns_sub_value PASSED [ 43%]
tests/test_fastnumbers.py::TestFastFloat::test_with_range_of_exponents_correctly_parses PASSED [ 44%]
tests/test_fastnumbers.py::TestFastFloat::test_given_float_string_returns_float PASSED [ 44%]
tests/test_fastnumbers.py::TestFastFloat::test_given_nan_string_returns_nan FAILED [ 45%]
tests/test_fastnumbers.py::TestFastFloat::test_given_inf_string_returns_inf PASSED [ 45%]
tests/test_fastnumbers.py::TestFastFloat::test_given_padded_float_strings_returns_float PASSED [ 45%]
tests/test_fastnumbers.py::TestFastFloat::test_given_padded_nan_returns_nan FAILED [ 46%]
tests/test_fastnumbers.py::TestFastFloat::test_given_padded_inf_string_returns_inf PASSED [ 46%]
tests/test_fastnumbers.py::TestFastFloat::test_given_int_returns_float PASSED [ 47%]
tests/test_fastnumbers.py::TestFastFloat::test_given_int_string_returns_float PASSED [ 47%]
tests/test_fastnumbers.py::TestFastFloat::test_given_padded_int_string_returns_float PASSED [ 48%]
tests/test_fastnumbers.py::TestFastFloat::test_given_unicode_digit_returns_float PASSED [ 48%]
tests/test_fastnumbers.py::TestFastFloat::test_given_unicode_numeral_returns_float PASSED [ 48%]
tests/test_fastnumbers.py::TestFastFloat::test_given_unicode_non_numeral_returns_as_is PASSED [ 49%]
tests/test_fastnumbers.py::TestFastFloat::test_given_unicode_of_more_than_one_char_returns_as_is PASSED [ 49%]
tests/test_fastnumbers.py::TestFastFloat::test_given_invalid_string_returns_string_as_is PASSED [ 50%]
tests/test_fastnumbers.py::TestFastFloat::test_given_invalid_string_raises_valueerror_if_raise_on_invalid_is_true PASSED [ 50%]
tests/test_fastnumbers.py::TestFastFloat::test_given_invalid_type_raises_typeerror PASSED [ 51%]
tests/test_fastnumbers.py::TestFastFloat::test_returns_default_value_if_given_invalid_string PASSED [ 51%]
tests/test_fastnumbers.py::TestFastFloat::test_raises_valueerror_if_raise_on_invalid_is_true_with_default PASSED [ 51%]
tests/test_fastnumbers.py::TestFastFloat::test_returns_input_as_is_if_valid_and_on_fail_is_given PASSED [ 52%]
tests/test_fastnumbers.py::TestFastFloat::test_returns_transformed_input_if_invalid_and_on_fail_is_given PASSED [ 52%]
tests/test_fastnumbers.py::TestFastInt::test_given_dumb_class_responds_to_internal_valueerror PASSED [ 53%]
tests/test_fastnumbers.py::TestFastInt::test_given_invalid_base_errors_with_valueerror PASSED [ 53%]
tests/test_fastnumbers.py::TestFastInt::test_given_float_returns_int PASSED [ 54%]
tests/test_fastnumbers.py::TestFastInt::test_given_nan_raises_valueerror_or_returns_as_is_or_returns_default PASSED [ 54%]
tests/test_fastnumbers.py::TestFastInt::test_given_inf_raises_overflowerror PASSED [ 54%]
tests/test_fastnumbers.py::TestFastInt::test_given_float_string_returns_string_as_is PASSED [ 55%]
tests/test_fastnumbers.py::TestFastInt::test_given_float_intlike_string_returns_string_as_is PASSED [ 55%]
tests/test_fastnumbers.py::TestFastInt::test_given_float_string_raises_valueerror_if_raise_on_invalid_is_true PASSED [ 56%]
tests/test_fastnumbers.py::TestFastInt::test_given_int_returns_int PASSED [ 56%]
tests/test_fastnumbers.py::TestFastInt::test_given_int_string_returns_int PASSED [ 57%]
tests/test_fastnumbers.py::TestFastInt::test_given_multiple_zeros_with_base_returns_zero PASSED [ 57%]
tests/test_fastnumbers.py::TestFastInt::test_given_padded_int_string_returns_int PASSED [ 57%]
tests/test_fastnumbers.py::TestFastInt::test_given_unicode_digit_returns_int PASSED [ 58%]
tests/test_fastnumbers.py::TestFastInt::test_given_unicode_numeral_returns_as_is PASSED [ 58%]
tests/test_fastnumbers.py::TestFastInt::test_given_unicode_non_numeral_returns_as_is PASSED [ 59%]
tests/test_fastnumbers.py::TestFastInt::test_given_unicode_of_more_than_one_char_returns_as_is PASSED [ 59%]
tests/test_fastnumbers.py::TestFastInt::test_given_invalid_string_returns_string_as_is PASSED [ 60%]
tests/test_fastnumbers.py::TestFastInt::test_given_invalid_string_raises_valueerror_if_raise_on_invalid_is_true PASSED [ 60%]
tests/test_fastnumbers.py::TestFastInt::test_given_invalid_type_raises_typeerror PASSED [ 60%]
tests/test_fastnumbers.py::TestFastInt::test_returns_default_value_if_given_invalid_string PASSED [ 61%]
tests/test_fastnumbers.py::TestFastInt::test_raises_valueerror_if_raise_on_invalid_is_true_with_default PASSED [ 61%]
tests/test_fastnumbers.py::TestFastInt::test_returns_input_as_is_if_valid_and_on_fail_is_given PASSED [ 62%]
tests/test_fastnumbers.py::TestFastInt::test_returns_transformed_input_if_invalid_and_on_fail_is_given PASSED [ 62%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_dumb_class_responds_to_internal_valueerror PASSED [ 63%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_float_returns_int PASSED [ 63%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_nan_raises_valueerror_or_returns_as_is_or_returns_default PASSED [ 63%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_inf_raises_overflowerror PASSED [ 64%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_float_string_returns_int PASSED [ 64%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_nan_string_raises_valueerror_with_raise_on_invalid_as_true FAILED [ 65%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_inf_string_raises_overflowerror_with_raise_on_invalid_as_true PASSED [ 65%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_padded_float_strings_returns_int PASSED [ 66%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_int_returns_int PASSED [ 66%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_int_string_returns_int PASSED [ 66%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_padded_int_string_returns_int PASSED [ 67%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_unicode_digit_returns_int PASSED [ 67%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_unicode_numeral_returns_int PASSED [ 68%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_unicode_non_numeral_returns_as_is PASSED [ 68%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_unicode_of_more_than_one_char_returns_as_is PASSED [ 69%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_invalid_string_returns_string_as_is PASSED [ 69%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_invalid_string_raises_valueerror_if_raise_on_invalid_is_true PASSED [ 69%]
tests/test_fastnumbers.py::TestFastForceInt::test_given_invalid_type_raises_typeerror PASSED [ 70%]
tests/test_fastnumbers.py::TestFastForceInt::test_returns_default_value_if_given_invalid_string PASSED [ 70%]
tests/test_fastnumbers.py::TestFastForceInt::test_raises_valueerror_if_raise_on_invalid_is_true_with_default PASSED [ 71%]
tests/test_fastnumbers.py::TestFastForceInt::test_returns_input_as_is_if_valid_and_on_fail_is_given PASSED [ 71%]
tests/test_fastnumbers.py::TestFastForceInt::test_returns_transformed_input_if_invalid_and_on_fail_is_given PASSED [ 72%]
tests/test_fastnumbers.py::TestIsReal::test_returns_true_if_given_int PASSED [ 72%]
tests/test_fastnumbers.py::TestIsReal::test_returns_true_if_given_float PASSED [ 72%]
tests/test_fastnumbers.py::TestIsReal::test_returns_false_if_given_int_and_str_only_is_true PASSED [ 73%]
tests/test_fastnumbers.py::TestIsReal::test_returns_false_if_given_float_and_str_only_is_true PASSED [ 73%]
tests/test_fastnumbers.py::TestIsReal::test_returns_true_if_given_int_string_padded_or_not PASSED [ 74%]
tests/test_fastnumbers.py::TestIsReal::test_returns_true_if_given_float_string_padded_or_not PASSED [ 74%]
tests/test_fastnumbers.py::TestIsReal::test_returns_false_if_given_string_and_num_only_is_true PASSED [ 75%]
tests/test_fastnumbers.py::TestIsReal::test_given_unicode_digit_returns_true PASSED [ 75%]
tests/test_fastnumbers.py::TestIsReal::test_given_unicode_numeral_returns_true PASSED [ 75%]
tests/test_fastnumbers.py::TestIsReal::test_given_unicode_non_numeral_returns_false PASSED [ 76%]
tests/test_fastnumbers.py::TestIsReal::test_given_unicode_of_more_than_one_char_returns_false PASSED [ 76%]
tests/test_fastnumbers.py::TestIsReal::test_returns_false_if_given_non_number_string PASSED [ 77%]
tests/test_fastnumbers.py::TestIsReal::test_returns_false_for_nan_string_unless_allow_nan_is_true PASSED [ 77%]
tests/test_fastnumbers.py::TestIsReal::test_returns_false_for_inf_string_unless_allow_infinity_is_true PASSED [ 78%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_false_if_given_int PASSED [ 78%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_true_if_given_float PASSED [ 78%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_false_if_given_float_and_str_only_is_true PASSED [ 79%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_true_if_given_int_string_padded_or_not PASSED [ 79%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_true_if_given_float_string_padded_or_not PASSED [ 80%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_false_if_given_string_and_num_only_is_true PASSED [ 80%]
tests/test_fastnumbers.py::TestIsFloat::test_given_unicode_digit_returns_true PASSED [ 81%]
tests/test_fastnumbers.py::TestIsFloat::test_given_unicode_numeral_returns_true PASSED [ 81%]
tests/test_fastnumbers.py::TestIsFloat::test_given_unicode_non_numeral_returns_false PASSED [ 81%]
tests/test_fastnumbers.py::TestIsFloat::test_given_unicode_of_more_than_one_char_returns_false PASSED [ 82%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_false_if_given_non_number_string PASSED [ 82%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_false_for_nan_string_unless_allow_nan_is_true PASSED [ 83%]
tests/test_fastnumbers.py::TestIsFloat::test_returns_false_for_inf_string_unless_allow_infinity_is_true PASSED [ 83%]
tests/test_fastnumbers.py::TestIsInt::test_returns_true_if_given_int PASSED [ 84%]
tests/test_fastnumbers.py::TestIsInt::test_returns_false_if_given_float PASSED [ 84%]
tests/test_fastnumbers.py::TestIsInt::test_returns_false_if_given_int_and_str_only_is_true PASSED [ 84%]
tests/test_fastnumbers.py::TestIsInt::test_returns_true_if_given_int_string_padded_or_not PASSED [ 85%]
tests/test_fastnumbers.py::TestIsInt::test_underscores PASSED            [ 85%]
tests/test_fastnumbers.py::TestIsInt::test_returns_false_if_given_float_string_padded_or_not PASSED [ 86%]
tests/test_fastnumbers.py::TestIsInt::test_returns_false_if_given_string_and_num_only_is_true PASSED [ 86%]
tests/test_fastnumbers.py::TestIsInt::test_given_unicode_digit_returns_true PASSED [ 87%]
tests/test_fastnumbers.py::TestIsInt::test_given_unicode_numeral_returns_false PASSED [ 87%]
tests/test_fastnumbers.py::TestIsInt::test_given_unicode_non_numeral_returns_false PASSED [ 87%]
tests/test_fastnumbers.py::TestIsInt::test_given_unicode_of_more_than_one_char_returns_false PASSED [ 88%]
tests/test_fastnumbers.py::TestIsInt::test_returns_false_if_given_non_number_string PASSED [ 88%]
tests/test_fastnumbers.py::TestIsInt::test_returns_false_for_nan_or_inf_string PASSED [ 89%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_true_if_given_int PASSED [ 89%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_false_if_given_non_integer_float PASSED [ 90%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_true_if_given_integer_float PASSED [ 90%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_false_if_given_int_and_str_only_is_true PASSED [ 90%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_false_if_given_integer_float_and_str_only_is_true PASSED [ 91%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_true_if_given_int_string_padded_or_not PASSED [ 91%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_true_if_given_integer_float_string_padded_or_not PASSED [ 92%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_false_if_given_non_integer_float_string_padded_or_not PASSED [ 92%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_false_if_given_string_and_num_only_is_true PASSED [ 93%]
tests/test_fastnumbers.py::TestIsIntLike::test_given_unicode_digit_returns_true PASSED [ 93%]
tests/test_fastnumbers.py::TestIsIntLike::test_given_unicode_non_digit_numeral_returns_false PASSED [ 93%]
tests/test_fastnumbers.py::TestIsIntLike::test_given_unicode_digit_numeral_returns_false PASSED [ 94%]
tests/test_fastnumbers.py::TestIsIntLike::test_given_unicode_non_numeral_returns_false PASSED [ 94%]
tests/test_fastnumbers.py::TestIsIntLike::test_given_unicode_of_more_than_one_char_returns_false PASSED [ 95%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_false_if_given_non_number_string PASSED [ 95%]
tests/test_fastnumbers.py::TestIsIntLike::test_returns_false_for_nan_or_inf_string PASSED [ 96%]
tests/test_fastnumbers_examples.py::test_fast_real FAILED                [ 96%]
tests/test_fastnumbers_examples.py::test_fast_float FAILED               [ 96%]
tests/test_fastnumbers_examples.py::test_fast_int PASSED                 [ 97%]
tests/test_fastnumbers_examples.py::test_fast_forceint FAILED            [ 97%]
tests/test_fastnumbers_examples.py::test_isreal PASSED                   [ 98%]
tests/test_fastnumbers_examples.py::test_isfloat PASSED                  [ 98%]
tests/test_fastnumbers_examples.py::test_isint PASSED                    [ 99%]
tests/test_fastnumbers_examples.py::test_isintlike PASSED                [ 99%]
tests/test_fastnumbers_examples.py::test_print_limits PASSED             [100%]

FASNUMBERS NUMERICAL LIMITS FOR THIS COMPILER BEFORE PYTHON FALLBACK:
MAXIMUM INTEGER LENTH: 18
MAX NUMBER FLOAT DIGITS: 11
MAXIMUM FLOAT EXPONENT: 99
MINIMUM FLOAT EXPONENT: -98


=================================== FAILURES ===================================
_________________________ InfNanTest.test_nan_from_str _________________________

self = <builtin_tests.py38.builtin_test_float.InfNanTest testMethod=test_nan_from_str>

    def test_nan_from_str(self):
>       self.assertTrue(isnan(float("nan")))
E       AssertionError: False is not true

tests/builtin_tests/py38/builtin_test_float.py:310: AssertionError
__________________________ InfNanTest.test_nan_signs ___________________________

self = <builtin_tests.py38.builtin_test_float.InfNanTest testMethod=test_nan_signs>

    @unittest.skipUnless(
        getattr(sys, "float_repr_style", "") == "short",
        "applies only when using short float repr style",
    )
    def test_nan_signs(self):
        # When using the dtoa.c code, the sign of float('nan') should
        # be predictable.
        self.assertEqual(copysign(1.0, float("nan")), 1.0)
>       self.assertEqual(copysign(1.0, float("-nan")), -1.0)
E       AssertionError: 1.0 != -1.0

tests/builtin_tests/py38/builtin_test_float.py:357: AssertionError
________________ TestFastReal.test_given_nan_string_returns_nan ________________

self = <test_fastnumbers.TestFastReal object at 0x652fc8244610>

    def test_given_nan_string_returns_nan(self):
>       assert math.isnan(fastnumbers.fast_real("nan"))
E       AssertionError: assert False
E        +  where False = <built-in function isnan>(0.0)
E        +    where <built-in function isnan> = math.isnan
E        +    and   0.0 = <built-in function fast_real>('nan')
E        +      where <built-in function fast_real> = fastnumbers.fast_real

tests/test_fastnumbers.py:245: AssertionError
________________ TestFastReal.test_given_padded_nan_returns_nan ________________

self = <test_fastnumbers.TestFastReal object at 0x652fcab765b0>

    def test_given_padded_nan_returns_nan(self):
>       assert math.isnan(fastnumbers.fast_real("     nan "))
E       AssertionError: assert False
E        +  where False = <built-in function isnan>(0.0)
E        +    where <built-in function isnan> = math.isnan
E        +    and   0.0 = <built-in function fast_real>('     nan ')
E        +      where <built-in function fast_real> = fastnumbers.fast_real

tests/test_fastnumbers.py:272: AssertionError
_______________ TestFastFloat.test_given_nan_string_returns_nan ________________

self = <test_fastnumbers.TestFastFloat object at 0x652fc81a9c10>

    def test_given_nan_string_returns_nan(self):
>       assert math.isnan(fastnumbers.fast_float("nan"))
E       AssertionError: assert False
E        +  where False = <built-in function isnan>(0.0)
E        +    where <built-in function isnan> = math.isnan
E        +    and   0.0 = <built-in function fast_float>('nan')
E        +      where <built-in function fast_float> = fastnumbers.fast_float

tests/test_fastnumbers.py:430: AssertionError
_______________ TestFastFloat.test_given_padded_nan_returns_nan ________________

self = <test_fastnumbers.TestFastFloat object at 0x652fc82293d0>

    def test_given_padded_nan_returns_nan(self):
>       assert math.isnan(fastnumbers.fast_float("     nan "))
E       AssertionError: assert False
E        +  where False = <built-in function isnan>(0.0)
E        +    where <built-in function isnan> = math.isnan
E        +    and   0.0 = <built-in function fast_float>('     nan ')
E        +      where <built-in function fast_float> = fastnumbers.fast_float

tests/test_fastnumbers.py:450: AssertionError
_ TestFastForceInt.test_given_nan_string_raises_valueerror_with_raise_on_invalid_as_true _

self = <test_fastnumbers.TestFastForceInt object at 0x652fc83ff460>

    def test_given_nan_string_raises_valueerror_with_raise_on_invalid_as_true(self):
        with raises(ValueError):
>           fastnumbers.fast_forceint("nan", raise_on_invalid=True)
E           Failed: DID NOT RAISE <class 'ValueError'>

tests/test_fastnumbers.py:765: Failed
________________________________ test_fast_real ________________________________

    def test_fast_real():
        # 1. float number
        assert fastnumbers.fast_real(-367.3268) == -367.3268
        assert fastnumbers.fast_real(-367.3268, raise_on_invalid=True) == -367.3268
        # 2. signed float string
        assert fastnumbers.fast_real("+367.3268") == +367.3268
        assert fastnumbers.fast_real("+367.3268", raise_on_invalid=True) == +367.3268
        # 3. float string with exponents
        assert fastnumbers.fast_real("-367.3268e207") == -367.3268e207
        assert fastnumbers.fast_real("1.175494351e-3810000000") == 0.0
        # 4. float string with padded whitespace
        assert fastnumbers.fast_real("   -367.04   ") == -367.04
        # 5. int number
        assert fastnumbers.fast_real(499) == 499
        # 6. signed int string
        assert fastnumbers.fast_real("-499") == -499
        # 7. int string with padded whitespace
        assert fastnumbers.fast_real("   +3001   ") == 3001
        # 8. long number
        assert fastnumbers.fast_real(35892482945872302493) == 35892482945872302493
        # 9. long string
        assert fastnumbers.fast_real("35892482945872302493") == 35892482945872302493
        # 10. return type
        assert isinstance(fastnumbers.fast_real(4029), int)
        assert isinstance(fastnumbers.fast_real(4029.0, coerce=False), float)
        assert isinstance(fastnumbers.fast_real(4029), int)
        assert isinstance(fastnumbers.fast_real(4029.0), int)
        assert isinstance(fastnumbers.fast_real(4029.5), float)
        assert isinstance(fastnumbers.fast_real("4029"), int)
        assert isinstance(fastnumbers.fast_real("4029.0"), int)
        assert isinstance(fastnumbers.fast_real("4029.0", coerce=False), float)
        # 11. TypeError for invalid input
        with raises(TypeError):
            fastnumbers.fast_real(["hey"])
        # 12. Invalid input string
        assert fastnumbers.fast_real("not_a_number") == "not_a_number"
        with raises(ValueError):
            assert fastnumbers.fast_real("not_a_number", raise_on_invalid=True)
        # 13. Invalid input string with numbers
        assert fastnumbers.fast_real("26.8 lb") == "26.8 lb"
        with raises(ValueError):
            assert fastnumbers.fast_real("26.8 lb", raise_on_invalid=True)
        # 14. Infinity
        assert fastnumbers.fast_real("inf") == float("inf")
        assert fastnumbers.fast_real("-iNFinity") == float("-inf")
        assert fastnumbers.fast_real("-iNFinity", inf=7608) == 7608
        # 15. NaN
>       assert math.isnan(fastnumbers.fast_real("nan"))
E       AssertionError: assert False
E        +  where False = <built-in function isnan>(0.0)
E        +    where <built-in function isnan> = math.isnan
E        +    and   0.0 = <built-in function fast_real>('nan')
E        +      where <built-in function fast_real> = fastnumbers.fast_real

tests/test_fastnumbers_examples.py:79: AssertionError
_______________________________ test_fast_float ________________________________

    def test_fast_float():
        # 1. float number
        assert fastnumbers.fast_float(-367.3268) == -367.3268
        assert fastnumbers.fast_float(-367.3268, raise_on_invalid=True) == -367.3268
        # 2. signed float string
        assert fastnumbers.fast_float("+367.3268") == +367.3268
        assert fastnumbers.fast_float("+367.3268", raise_on_invalid=True) == +367.3268
        # 3. float string with exponents
        assert fastnumbers.fast_float("-367.3268e27") == -367.3268e27
        assert fastnumbers.fast_float("-367.3268E27") == -367.3268e27
        assert fastnumbers.fast_float("-367.3268e207") == -367.3268e207
        assert fastnumbers.fast_float("1.175494351E-3810000000") == 0.0
        # 4. float string with padded whitespace
        assert fastnumbers.fast_float("   -367.04   ") == -367.04
        # 5. int number
        assert fastnumbers.fast_float(499) == 499.0
        # 6. signed int string
        assert fastnumbers.fast_float("-499") == -499.0
        # 7. int string with padded whitespace
        assert fastnumbers.fast_float("   +3001   ") == 3001
        # 8. long number
        assert fastnumbers.fast_float(35892482945872302493) == 35892482945872302493.0
        # 9. long string
        assert fastnumbers.fast_float("35892482945872302493") == 35892482945872302493.0
        # 10. return type
        assert isinstance(fastnumbers.fast_float(4029), float)
        assert isinstance(fastnumbers.fast_float("4029"), float)
        # 11. TypeError for invalid input
        with raises(TypeError):
            fastnumbers.fast_float(["hey"])
        # 12. Invalid input string
        assert fastnumbers.fast_float("not_a_number") == "not_a_number"
        with raises(ValueError):
            assert fastnumbers.fast_float("not_a_number", raise_on_invalid=True)
        # 13. Invalid input string with numbers
        assert fastnumbers.fast_float("26.8 lb") == "26.8 lb"
        with raises(ValueError):
            assert fastnumbers.fast_float("26.8 lb", raise_on_invalid=True)
        # 14. Infinity
        assert fastnumbers.fast_float("inf") == float("inf")
        assert fastnumbers.fast_float("-iNFinity") == float("-inf")
        assert fastnumbers.fast_float("-iNFinity", inf=523) == 523
        # 15. NaN
>       assert math.isnan(fastnumbers.fast_float("nAn"))
E       AssertionError: assert False
E        +  where False = <built-in function isnan>(0.0)
E        +    where <built-in function isnan> = math.isnan
E        +    and   0.0 = <built-in function fast_float>('nAn')
E        +      where <built-in function fast_float> = fastnumbers.fast_float

tests/test_fastnumbers_examples.py:147: AssertionError
______________________________ test_fast_forceint ______________________________

    def test_fast_forceint():
        # 1. float number
        assert fastnumbers.fast_forceint(-367.3268) == -367
        assert fastnumbers.fast_forceint(-367.3268, raise_on_invalid=True) == -367
        # 2. signed float string
        assert fastnumbers.fast_forceint("+367.3268") == 367
        assert fastnumbers.fast_forceint("+367.3268", raise_on_invalid=True) == 367
        # 3. float string with exponents
        assert fastnumbers.fast_forceint("-367.3268e207") == int(-367.3268e207)
        # 4. float string with padded whitespace
        assert fastnumbers.fast_forceint("   -367.04   ") == -367
        # 5. int number
        assert fastnumbers.fast_forceint(499) == 499
        # 6. signed int string
        assert fastnumbers.fast_forceint("-499") == -499
        # 7. int string with padded whitespace
        assert fastnumbers.fast_forceint("   +3001   ") == 3001
        # 8. long number
        assert fastnumbers.fast_forceint(35892482945872302493) == 35892482945872302493
        # 9. long string
        assert fastnumbers.fast_forceint("35892482945872302493") == 35892482945872302493
        # 10. return type
        assert isinstance(fastnumbers.fast_forceint(4029.00), int)
        assert isinstance(fastnumbers.fast_forceint("4029.00"), int)
        # 11. TypeError for invalid input
        with raises(TypeError):
            fastnumbers.fast_forceint(["hey"])
        # 12. Invalid input string
        assert fastnumbers.fast_forceint("not_a_number") == "not_a_number"
        with raises(ValueError):
            assert fastnumbers.fast_forceint("not_a_number", raise_on_invalid=True)
        # 13. Invalid input string with numbers
        assert fastnumbers.fast_forceint("26.8 lb") == "26.8 lb"
        with raises(ValueError):
            assert fastnumbers.fast_forceint("26.8 lb", raise_on_invalid=True)
        # 14. Infinity
        assert fastnumbers.fast_forceint("inf") == "inf"
        assert fastnumbers.fast_forceint("-iNFinity") == "-iNFinity"
        # 15. NaN
>       assert fastnumbers.fast_forceint("nan") == "nan"
E       AssertionError: assert 0 == 'nan'
E         +0
E         -'nan'

tests/test_fastnumbers_examples.py:273: AssertionError
=========================== short test summary info ============================
FAILED tests/test_builtins.py::InfNanTest::test_nan_from_str - AssertionError...
FAILED tests/test_builtins.py::InfNanTest::test_nan_signs - AssertionError: 1...
FAILED tests/test_fastnumbers.py::TestFastReal::test_given_nan_string_returns_nan
FAILED tests/test_fastnumbers.py::TestFastReal::test_given_padded_nan_returns_nan
FAILED tests/test_fastnumbers.py::TestFastFloat::test_given_nan_string_returns_nan
FAILED tests/test_fastnumbers.py::TestFastFloat::test_given_padded_nan_returns_nan
FAILED tests/test_fastnumbers.py::TestFastForceInt::test_given_nan_string_raises_valueerror_with_raise_on_invalid_as_true
FAILED tests/test_fastnumbers_examples.py::test_fast_real - AssertionError: a...
FAILED tests/test_fastnumbers_examples.py::test_fast_float - AssertionError: ...
FAILED tests/test_fastnumbers_examples.py::test_fast_forceint - AssertionErro...
================== 10 failed, 222 passed, 1 skipped in 13.77s ==================

```

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,6 +7,7 @@ python:
   - 3.6
   - 3.7
   - 3.8
+  - 3.9
 
 env:
   - CC=clang
@@ -17,8 +18,9 @@ jobs:
     # Windows tests!
     - {python: 3.5, os: windows, env: PYTHON_VERSION=3.5.4, language: sh}
     - {python: 3.6, os: windows, env: PYTHON_VERSION=3.6.8, language: sh}
-    - {python: 3.7, os: windows, env: PYTHON_VERSION=3.7.4, language: sh}
-    - {python: 3.8, os: windows, env: PYTHON_VERSION=3.8.0, language: sh}
+    - {python: 3.7, os: windows, env: PYTHON_VERSION=3.7.9, language: sh}
+    - {python: 3.8, os: windows, env: PYTHON_VERSION=3.8.6, language: sh}
+    - {python: 3.9, os: windows, env: PYTHON_VERSION=3.9.0, language: sh}
 
     # This "code quality" stage does static analysis and formatting checks.
     # Platform- and Python-version-independent.
@@ -121,3 +123,4 @@ cache:
     - /c/Python36
     - /c/Python37
     - /c/Python38
+    - /c/Python39

--- a/.travis.yml
+++ b/.travis.yml
@@ -63,28 +63,28 @@ jobs:
       name: "Build and Deploy Linux Wheels"
       python: 3.6
       services: docker
-      env: CIBW_SKIP="cp27-*"
+      env: CIBW_BUILD="cp3*"
       before_install: skip
-      install: python -m pip install twine cibuildwheel==1.0.0
+      install: python -m pip install twine cibuildwheel==1.6.4
       before_script: if [ -z $TRAVIS_TAG ]; then export TWINE_REPOSITORY_URL=https://test.pypi.org/legacy/; fi
       script: python -m cibuildwheel --output-dir wheelhouse
       after_success: python -m twine upload --user SethMMorton --skip-existing wheelhouse/*.whl
     - stage: deploy
       name: "Build and Deploy MacOS Wheels"
       os: osx
-      env: CIBW_SKIP="cp27-*"
+      env: CIBW_BUILD="cp3*"
       language: sh
       before_install: skip
-      install: python -m pip install twine cibuildwheel==1.0.0
+      install: python3 -m pip install twine cibuildwheel==1.6.4
       before_script: if [ -z $TRAVIS_TAG ]; then export TWINE_REPOSITORY_URL=https://test.pypi.org/legacy/; fi
-      script: python -m cibuildwheel --output-dir wheelhouse
-      after_success: python -m twine upload --user SethMMorton --skip-existing wheelhouse/*.whl
+      script: python3 -m cibuildwheel --output-dir wheelhouse
+      after_success: python3 -m twine upload --user SethMMorton --skip-existing wheelhouse/*.whl
     - stage: deploy
       name: "Build and Deploy Windows Wheels"
       os: windows
-      env: [PYTHON_VERSION=3.6.8, CIBW_SKIP="cp27-*"]
+      env: [PYTHON_VERSION=3.6.8, CIBW_BUILD="cp3*"]
       language: sh
-      install: python -m pip install twine cibuildwheel==1.0.0
+      install: python -m pip install twine cibuildwheel==1.6.4
       before_script: if [ -z $TRAVIS_TAG ]; then export TWINE_REPOSITORY_URL=https://test.pypi.org/legacy/; fi
       script: python -m cibuildwheel --output-dir wheelhouse
       after_success: python -m twine upload --user SethMMorton --skip-existing wheelhouse/*.whl

--- a/.travis.yml
+++ b/.travis.yml
@@ -20,7 +20,7 @@ jobs:
     - {python: 3.6, os: windows, env: PYTHON_VERSION=3.6.8, language: sh}
     - {python: 3.7, os: windows, env: PYTHON_VERSION=3.7.9, language: sh}
     - {python: 3.8, os: windows, env: PYTHON_VERSION=3.8.6, language: sh}
-    - {python: 3.9, os: windows, env: PYTHON_VERSION=3.9.0, language: sh}
+    #- {python: 3.9, os: windows, env: PYTHON_VERSION=3.9.0, language: sh}
 
     # This "code quality" stage does static analysis and formatting checks.
     # Platform- and Python-version-independent.

--- a/dev/check-formatting.py
+++ b/dev/check-formatting.py
@@ -33,6 +33,7 @@ print(astyle_c_ret.stdout)
 astyle_h = astyle + [
     "--exclude=docstrings.h",
     "--exclude=fn_bool.h",
+    "--exclude=options.h",
     "--exclude=pstdint.h",
     "include/fastnumbers/*.h",
 ]

--- a/include/fastnumbers/numbers.h
+++ b/include/fastnumbers/numbers.h
@@ -14,6 +14,13 @@
 extern "C" {
 #endif
 
+/*
+ * Copy of _Py_dg_stdnan from the Python code base.
+ * Needed to generate NaN on Windows.
+ */
+double
+compat_generate_nan(int sign);
+
 /* All the awesome MACROS */
 
 #define PyNumber_IsNAN(pynum) (PyFloat_Check(pynum) && Py_IS_NAN(PyFloat_AS_DOUBLE(pynum)))
@@ -21,7 +28,7 @@ extern "C" {
 
 /* Return NaN correctly on this system. */
 #if !defined(PY_NO_SHORT_FLOAT_REPR)
-#define PyFloat_from_NaN(negative) PyFloat_FromDouble(_Py_dg_stdnan(negative));
+#define PyFloat_from_NaN(negative) PyFloat_FromDouble(compat_generate_nan(negative));
 #else
 #define PyFloat_from_NaN(negative) \
     (negative) ? PyFloat_FromDouble(-Py_NAN) : PyFloat_FromDouble(Py_NAN);

--- a/setup.cfg
+++ b/setup.cfg
@@ -28,6 +28,7 @@ classifiers =
 	Programming Language :: Python :: 3.6
 	Programming Language :: Python :: 3.7
 	Programming Language :: Python :: 3.8
+	Programming Language :: Python :: 3.9
 	Topic :: Scientific/Engineering :: Information Analysis
 	Topic :: Utilities
 	Topic :: Text Processing

--- a/src/numbers.c
+++ b/src/numbers.c
@@ -19,6 +19,38 @@
 #endif
 
 
+/*
+ * Copy of _Py_dg_stdnan from the Python code base.
+ * Dynamically generate an NaN. Needed for Windows.
+ * I hate copy/pasting code, but this seems like the only way.
+ */
+typedef union {
+    double d;
+    uint32_t L[2];
+} U;
+#ifdef DOUBLE_IS_LITTLE_ENDIAN_IEEE754
+#define word0(x) (x)->L[1]
+#define word1(x) (x)->L[0]
+#else
+#define word0(x) (x)->L[0]
+#define word1(x) (x)->L[1]
+#endif
+#define dval(x) (x)->d
+#define NAN_WORD0 0x7ff80000
+#define NAN_WORD1 0
+#define Sign_bit 0x80000000
+double
+compat_generate_nan(int sign) {
+    U rv;
+    word0(&rv) = NAN_WORD0;
+    word1(&rv) = NAN_WORD1;
+    if (sign) {
+        word0(&rv) |= Sign_bit;
+    }
+    return dval(&rv);
+}
+
+
 static bool
 _PyFloat_is_Intlike(PyObject *obj) {
     /* NOTE: This code copy/pasted with modification from the CPython source.

--- a/tests/builtin_support.py
+++ b/tests/builtin_support.py
@@ -92,10 +92,10 @@ def _parse_guards(guards):
 # or to run them only on the implementation(s) guarded by the arguments.
 def check_impl_detail(**guards):
     """This function returns True or False depending on the host platform.
-       Examples:
-          if check_impl_detail():               # only on CPython (default)
-          if check_impl_detail(jython=True):    # only on Jython
-          if check_impl_detail(cpython=False):  # everywhere except on CPython
+    Examples:
+       if check_impl_detail():               # only on CPython (default)
+       if check_impl_detail(jython=True):    # only on Jython
+       if check_impl_detail(cpython=False):  # everywhere except on CPython
     """
     guards, default = _parse_guards(guards)
     return guards.get(platform.python_implementation().lower(), default)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -4,5 +4,5 @@ import hypothesis
 # For some reason it thinks that the text/binary generation is too
 # slow then causes the tests to fail.
 hypothesis.settings.register_profile(
-    "slow-tests", suppress_health_check=[hypothesis.HealthCheck.too_slow],
+    "slow-tests", suppress_health_check=[hypothesis.HealthCheck.too_slow]
 )

--- a/tox.ini
+++ b/tox.ini
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py35, py36, py37, py38
+envlist = py35, py36, py37, py38, py39
 # Other valid environments are:
 #   format
 #   format-check

